File: DSA/DSA Using Go/Shell_Sort.go
package main

import "fmt"

func ShellSort(slice []int) {
	var n, gap, i, temp, j int
	n = len(slice)
	gap = n / 2
	for gap > 0 {
		i = gap
		for i < n {
			temp = slice[i]
			j = i - gap
			for j >= 0 && slice[j] > temp {
				slice[j+gap] = slice[j]
				j = j - gap
			}
			slice[j+gap] = temp
			i++
		}
		gap = gap / 2
	}
}

func main() {
	var slice = []int{3, 5, 8, 9, 6, 2}
	fmt.Println("Before sorting:", slice)
	ShellSort(slice)
	fmt.Println("After sorting:", slice)
}

File: DSA/DSA Using Python/Shell_Sort.py
def ShellSort(list):
    n = len(list)
    gap=n//2
    while gap > 0:
        i = gap
        while i < n:
            temp = list[i]
            j = i - gap
            while j>=0 and list[j]>temp:
                list[j+gap]=list[j]
                j = j-gap
            list[j+gap]=temp
            i += 1
        gap = gap // 2

list = [3,5,8,9,6,2]
print('Original Array:',list)
ShellSort(list)
print('Sorted Array:',list)